src/sas/sasgui/perspectives/fitting/fitpage.py

```diff
@@ -1238,7 +1238,12 @@ def _on_select_model(self, event=None):
             custom_model = CUSTOM_MODEL
             mod_cat = self.categorybox.GetStringSelection()
             if mod_cat == custom_model:
+                ################This is where the model dies###############
+                temp_id = self.model.id
                 temp = self.parent.update_model_list()
+                for v in self.parent.model_dictionary.values():
```

Comment:
How about:
```
for k in self.parent.model_dictionary:
    v = self.parent.model_dictionary[k]
    if v.id == temp_id:
        self.model = v()
```
the reason being that in Python 2.x asking for dictionary values returns a list, which can be inefficient. If this isn't a problem then perhaps the way you have it is better.